vm/src/class.rs

```diff
@@ -10,9 +12,15 @@ use crate::{
 };
 use rustpython_common::static_cell;
 
-pub trait StaticType {
+/// A type that represents a statically-allocated Python class.
+///
+/// # Safety
+///
+/// `type_id` must be accurate.
```

Comment:
I understand wrong type_id can cause a bug, but not sure how it can make unsafety. Do you have a scenario?